client.go

```diff
@@ -185,7 +200,9 @@ func (c *Client) RegisterPeer(addr wire.Address, host, alias string) (Peer, erro
 
 	peer := Peer{ethAddr, alias}
 	c.peers[ethAddr] = peer
-	c.dialer.Register(addr, host)
+	if c.registerInDialer != nil {
```

Comment:
This looks ugly. You could define a custom dialer interface that only allows dialers that have a register method. Then you wouldn't need to introduce a separate variable for the register functionality.

On the other hand, I wonder if we even need to support a register method on the payment client object. We could define a type that contains all the necessary information about a peer together, and the payment client extracts the needed information on the fly.

Reply:
Yes i also did not like having this function there with a `nil` check. It is only needed when the user does not provide a `Dialer` with `WithDialer`, hence the check. Defining a new `Dialer` interface would therefore restrict to much for the user.  

I like the thought about not having a `RegisterPeer` method.

Reply:
I added a `Dialer` interface now with a `Register` function.